elastic/cloud-on-k8s
# Ignore resources reconciled by older controllers

Resolves https://github.com/elastic/cloud-on-k8s/issues/1241

* If there is no existing annotation on a resource, it may be old (previously reconciled by a <0.9.0 controller), or it may be brand new. We update the annotation indicating which one by looking if there are any pods that belong to the CR already
* If there is an existing annotation, we make sure it's at least a 0.9.0 pre-release version before continuing

Still need to do some manual testing but so far looks good. Opening up for early reviews now since there's a bit of time pressure.

Manual testing steps I used:
checked out 0.8.1 tag, `dep ensure --vendor-only && make run`, ran `make samples` and let it build the resources successfully.
checked out this branch, `dep ensure --vendor-only && make run`. Verified that it didn't act on the existing resources and added an annotation with `0.8.0-UNKNOWN`.
